### DISCUSSION TITLE
Abseil not using Overlay port

### DISCUSSION THREAD

**Discussion by diehard2** [microsoft/vcpkg#31847] [General]
I've created an overlay port that applies several compilation and linker options to all of my vcpkg libraries. However, Abseil isn't respecting this and is defaulting to x64-linux

``` 
cmake 3.25.1
features core
fix-32-bit-arm.patch 60794fddcdd62e554126ba98536d6990015be93ce156245dc63a1dd597b31c95
fix-dll-support.patch 9d8c44fb816ef6f1547260391e58d2bfc9af9dc2a44f6874b2a7fb2c45c5bb03
portfile.cmake 509eaf76b34d802ef0beb0f843d2463b782913ec913a7a9e74c997a28f6ad66b
ports.cmake 5a8e00cedff0c898b1f90f7d129329d0288801bc9056562b039698caf31ff3f3
post_build_checks 2
triplet x64-linux
```

for instance, arrow is showing what I would expect

```
post_build_checks 2
protobuf 144ccec6d941f11a11097f0514747c81f3ad804ed93e3617d7097c0740b8199e
rapidjson 515dc6f1c50c422992b26a1483170052b37c4f892dd83da554742bd7a53c38af
re2 2b16c270808ca7abd2805fa0500b16fa97a0d37309c050a191156fa9a8d64e5e
snappy 683adc67be6b6655755e4785785a220db57349461ea2c41783a2e70549e430d5
thrift 09a063e437c2fa79a9cd920c577c6e89c61659d622d48ffca2228d31b081425f
thrift.patch 37e023442317e56582b4ef3d75cda8bad88d626cb8fee4eb0b7cd61a7662c1c6
triplet x64-linux-dynamic
```

Now, this isn't the end of the world other than I can't get Abseil to link when I'm using flto and I'm guessing its because the options are different. I've tried adding the following in CMakeLists.txt before the toolchain file directive to no avail. How can I force every library to build using a custom overlay file from the CMakeLists.txt? Even if I could specify the static overlay that would be fine (but I need to keep dynamic on for most of them). Thanks for any help

My overlay directives look like this

set(VCPKG_OVERLAY_TRIPLETS ../custom-triplets)
set(VCPKG_DEFAULT_HOST_TRIPLET x64-linux-dynamic-mine)
set(VCPKG_TARGET_TRIPLET x64-linux-dynamic-mine)

**Comment by Osyotr** [comment]
Probably `VCPKG_HOST_TRIPLET` should be used instead of `VCPKG_DEFAULT_HOST_TRIPLET`.
